### DISCUSSION TITLE
SSH certificate support in SSH tunnel

### DISCUSSION THREAD

**Discussion by arghaya** [dbeaver/dbeaver#14580] [Q&A] [answered]
Hi Team, 

I am using an SSH certificate to access my servers. But there is no option to add SSH certificate, in SSH tunnel tab there is an option to add public key but it only accepts private key file to authenticate. On selecting the certificate file, I am getting an error: 

 _Can't initialize tunnel
  Cannot establish tunnel
  Cannot establish tunnel
    invalid privatekey: [B@5ecde315_

On terminal I am able to ssh using:  **ssh -i id-cert.pub username@servername**

It would be really nice to have the feature embedded out-of-the-box.

Is there any workaround for this, as I am stuck.

Thanks in advance.

**Comment by uslss** [comment]
thanks for suggestion

**Comment by mahajanankur** [comment]
Really a great feature to have. This will provide much more security to the SSH tunnelling.

@arghaya, thanks for sharing!!

**Comment by RajeshJangid** [comment, marked as answer]
I also could not find accurate solution from dbeaver, but following hack should help you solve the issue

execute the command from terminal

**ssh -i id-cert.pub -fNL 3308:localhost:3307  username@server.com**

In the above command 
1. replace 3308 for local port through which all requests should be served
2. 3307 is the port where service on remote server is listening

So after that in Dbeaver you can configure using localhost and port used for tunnelling.

**Comment by arghaya** [comment]
Thanks @RajeshJangid  works like a charm.

**Comment by mahajanankur** [comment]
@RajeshJangid, you ROCK!!
